frontend/src/App.tsx

```diff
@@ -741,18 +741,20 @@ export class App extends PureComponent<Props, State> {
     )?.pageName as string
     const viewingMainPage = newPageScriptHash === mainPage.pageScriptHash
 
-    const baseUriParts = this.getBaseUriParts()
-    if (baseUriParts) {
-      const { basePath } = baseUriParts
-      const queryString = this.getQueryString()
+    if (this.state.currentPageScriptHash !== newPageScriptHash) {
```

Comment:
One small thing that I think we'll want to clean up while making changes to this method:

We currently already destructure `currentPageScriptHash` from  `this.state` on line 729 above, but it's only used at the very end of the method. I think it would be more clear to rename the variable to `prevPageScriptHash` when destructuring it and using it both here and at the end of the method

Reply:
Thank you! I fixed the code.
* Using the destructured variable:  [8a7318e](https://github.com/streamlit/streamlit/pull/6266/commits/8a7318ea433ccf0ee079e864c08fdf1b9829346e)
* Rename the variable: [d077d97](https://github.com/streamlit/streamlit/pull/6266/commits/d077d978cdd09e08c7dec1b2f658cbac1084c1be)
  * BTW, I think `current` and `next` already can make sense though. What do you think?

Reply:
> BTW, I think current and next already can make sense though. What do you think?

I think `current` and `next` is already clear (especially in the context of the rest of the code) but just a bit less accurate than `prev` and `new` because by the time this code is being run, it's `newPageScriptHash` that corresponds to the script currently running on the server.

Definitely a small/inconsequential difference either way, though.